cfme/tests/infrastructure/test_rest_automation_request.py

```diff
@@ -1,123 +1,295 @@
+        """Tests approving automation requests using /api/requests.
+
+        Metadata:
+            test_flag: rest, requests
+        """
+        approve_requests(collection, rest_api, requests_pending, from_detail)
+
+    @pytest.mark.uncollectif(lambda: current_version() < '5.7')
+    @pytest.mark.tier(3)
+    @pytest.mark.parametrize(
+        'from_detail', [True, False],
+        ids=['from_detail', 'from_collection'])
+    def test_deny_requests(self, collection, rest_api, requests_pending, from_detail):
+        """Tests denying automation requests using /api/requests.
+
+        Metadata:
+            test_flag: rest, requests
+        """
+        deny_requests(collection, rest_api, requests_pending, from_detail)
+
+    @pytest.mark.uncollectif(lambda: current_version() < '5.7')
+    @pytest.mark.tier(3)
+    @pytest.mark.parametrize(
+        'from_detail', [True, False],
+        ids=['from_detail', 'from_collection'])
+    def test_edit_requests(self, collection, rest_api, requests_pending, from_detail):
+        """Tests editing requests using /api/requests.
+
+        Metadata:
+            test_flag: rest, requests
+        """
+        body = {'options': {'arbitrary_key_allowed': 'test_rest'}}
+
+        if from_detail:
+            if BZ('1418331', forced_streams=['5.7', 'upstream']).blocks:
+                pytest.skip("Affected by BZ1418331, cannot test.")
+            for request in requests_pending:
+                request.action.edit(**body)
+                assert rest_api.response.status_code == 200
+        else:
+            identifiers = []
+            for i, resource in enumerate(requests_pending):
+                loc = ({'id': resource.id}, {'href': '{}/{}'.format(collection._href, resource.id)})
+                identifiers.append(loc[i % 2])
+            collection.action.edit(*identifiers, **body)
+            assert rest_api.response.status_code == 200
+
+        for request in requests_pending:
+            request.reload()
+            assert request.options['arbitrary_key_allowed'] == 'test_rest'
+
+    @pytest.mark.uncollectif(lambda: current_version() < '5.7')
+    def test_create_requests_parallel(self, rest_api):
+        """Create automation requests in parallel.
+
+        Metadata:
+            test_flag: rest, requests
+        """
+        output = mp.Queue()
+        entry_point = rest_api._entry_point
+        auth = rest_api._auth
+
+        def _gen_automation_requests(output):
+            api = MiqApi(entry_point, auth, verify_ssl=False)
+            requests_data = _automation_requests_data(
+                'nonexistent_vm', requests_collection=True, approve=False)
+            api.collections.requests.action.create(*requests_data[:2])
+            result = (api.response.status_code, api.response.json())
+            output.put(result)
+
+        processes = [
+            mp.Process(target=_gen_automation_requests, args=(output,))
```

Comment:
I wanted to make sure that it would be really parallel. With threads it might not be the case due to GIL.